execution/process.go

```diff
@@ -54,16 +59,43 @@ func (p *ProcessByKey) AddRecord(ctx context.Context, tx storage.StateTransactio
 	}
 
 	keyTuple := octosql.MakeTuple(key)
-	err = p.processFunction.AddRecord(ctx, tx, inputIndex, keyTuple, record)
-	if err != nil {
-		return errors.Wrap(err, "couldn't add record to process function")
-	}
 
 	eventTime := MaxWatermark
+	eventTimePrefixedTx := tx
 	if len(p.eventTimeField) > 0 {
 		eventTime = record.EventTime().AsTime()
+
+		// Adding record event time to keyTuple so that during Trigger prefixing process function is possible
+		keyTuple = octosql.MakeTuple(append([]octosql.Value{octosql.MakeTime(eventTime)}, key...))
+
+		// Adding keyTuple to event time storage
+		eventTimeMap := storage.NewMap(tx.WithPrefix(eventTimesSeenPrefix))
+
+		octoEventTime := octosql.MakeTime(eventTime)
```

Comment:
This is not entirely clear. You could add a comment explaining it a bit more like "for every event time, we store keys that appeared in that event time" or smth like that